ci/kind/kind-setup.sh

```diff
@@ -244,25 +245,66 @@ function configure_extra_networks {
   done
 }
 
+# add_kind_ipam_routes add routes for non-ipam test-pods.
+function add_kind_ipam_routes {
+  kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{" "}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' |\
+  while read pod_cidr node_ip; do
+    docker_run_with_host_net ip route add "$pod_cidr" via "$node_ip"
+  done
+}
+
+# delete_kind_ipam_routes delete the routes of non-ipam test-pods from the Kind testbed.
+function delete_kind_ipam_routes { 
+  cluster=$(kind get clusters)
+  if [ ! -z "$cluster" ]; then
+    cidrs=$(kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')
+    echo -e "$cidrs" | tr ' ' '\n'| while read cidr; do
+      del_routes=$(ip route | awk -v cidr="$cidr" '$0 ~ cidr {print$1}')
+      if [[ -n $del_routes ]]; then
+        docker_run_with_host_net ip route del $del_routes || true
```

Comment:
Why doesn't it used the same way as `add_kind_ipam_routes` to construct the route delete command but write more complex code?

Reply:
in the flexible-ipam builder first we are deleting cluster , specific ipam routes, vlans..etc. for the routes we need podCIDR to check routes are present on testbed or not and for that we need to check if any cluster are there if any cluster there then use `kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}')` get podCIDR from any kind cluster. then delete the routes it.


to check routes, I need cidr and for cidr, I have to use kubectl... cmd and if cluster is not present then kubectl... cmd show the error and exit the script with fail.

`$ kubectl get nodes -o jsonpath='{.items[*].spec.podCIDR}'
E0808 12:19:00.813784 2299019 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0808 12:19:00.814189 2299019 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0808 12:19:00.815623 2299019 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0808 12:19:00.815860 2299019 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0808 12:19:00.817192 2299019 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?`


could you please suggest me any other way to do it?